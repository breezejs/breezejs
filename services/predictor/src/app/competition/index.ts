/**
 * Breeze predictor competition
 * @module breeze/predictor/competition
 */
export * from './competition.controller';
export * from './competition.dto';
export * from './competition.entity';
export * from './competition.enum';
export * from './competition.module';
export * from './competition.providers';
export * from './competition.service';
