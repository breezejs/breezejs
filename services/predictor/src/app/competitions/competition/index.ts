/**
 * Breeze predictor service competition module
 * @module breeze/predictor/competitions/competition
 */
export * from './competition.controller';
