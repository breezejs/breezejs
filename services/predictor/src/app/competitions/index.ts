/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
export * from './competitions.module';
