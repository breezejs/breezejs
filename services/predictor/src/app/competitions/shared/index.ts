/**
 * Breeze predictor service shared competitions module
 * @module breeze/predictor/competitions/shared
 */
export * from './competition.dto';
