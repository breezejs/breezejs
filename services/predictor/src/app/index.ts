/**
 * Breeze BB predictor
 * @module breeze-bb/predictor
 */
export * from './app.module';
