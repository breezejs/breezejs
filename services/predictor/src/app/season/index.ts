/**
 * Breeze predictor season
 * @module breeze/predictor/season
 */
export * from './season.controller';
export * from './season.dto';
export * from './season.entity';
export * from './season.enum';
export * from './season.module';
export * from './season.providers';
export * from './season.service';
