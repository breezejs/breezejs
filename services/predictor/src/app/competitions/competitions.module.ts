/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
import {CompetitionModule} from '@breeze/competition';
import {Module} from '@nestjs/common';
import {CompetitionController} from './competition';

/**
 * Competitions module for predictor service
 */
@Module({
  controllers: [
    CompetitionController
  ],
  imports: [
    CompetitionModule
  ]
})
export class CompetitionsModule {}
