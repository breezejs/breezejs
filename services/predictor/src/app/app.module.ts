/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
import {Module} from '@nestjs/common';
import {SeasonModule} from './season';
import {TeamModule} from './team';

/**
 * Predictor service app module
 */
@Module({
  imports: [
    SeasonModule,
    TeamModule
  ]
})
export class AppModule {}
