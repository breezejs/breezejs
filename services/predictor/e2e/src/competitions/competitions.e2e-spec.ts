/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
import {INestApplication, ValidationPipe} from '@nestjs/common';
import {Test, TestingModule} from '@nestjs/testing';
import * as supertest from 'supertest';
import {CompetitionsModule} from '../../../src/app/competitions';

describe('Competitions', (): void => {
  let app: INestApplication;
  let id: string;
  let request: supertest.SuperTest<supertest.Request>;

  beforeAll(async (): Promise<void> => {
    const testModule: TestingModule = await Test
      .createTestingModule({
        imports: [
          CompetitionsModule
        ]
      })
      .compile();

    app = testModule.createNestApplication();

    app.useGlobalPipes(new ValidationPipe());

    await app.init();

    request = supertest(app.getHttpServer());
  });

  afterAll(async (): Promise<void> => {
    await app.close();
  });

  describe('[GET] /competitions', (): void => {
    describe('with no filtering or sorting', (): void => {
      it('should list first page of competitions with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/competitions');

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should list second page of competitions with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/competitions').query({
          page: 2
        });

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should return no content if out of bounds', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/competitions').query({
          page: 3
        });

        expect(status).toEqual(204);
        expect(body).toMatchSnapshot();
      });
    });

    describe('with filtering', (): void => {
      it('should list first page of competitions with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/competitions').query({
          fields: 'competition.id,competition.name'
        });

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should list second page of competitions with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/competitions').query({
          fields: 'competition.id,competition.name',
          page: 2
        });

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should return no content if out of bounds', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/competitions').query({
          fields: 'competition.id,competition.name',
          page: 3
        });

        expect(status).toEqual(204);
        expect(body).toMatchSnapshot();
      });

      it('should return a bad request if filtered field does not exist', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/competitions').query({
          fields: 'competition.club'
        });

        expect(status).toEqual(400);
        expect(body).toMatchSnapshot();
      });
    });

    describe('with sorting', (): void => {
      describe('ascending', (): void => {
        it('should list first page of competitions with all values', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/competitions').query({
            sort: 'competition.name'
          });

          expect(status).toEqual(200);
          expect(body).toMatchSnapshot();
        });

        it('should list second page of competitions with all values', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/competitions').query({
            page: 2,
            sort: 'competition.name'
          });

          expect(status).toEqual(200);
          expect(body).toMatchSnapshot();
        });

        it('should return no content if out of bounds', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/competitions').query({
            page: 3,
            sort: 'competition.name'
          });

          expect(status).toEqual(204);
          expect(body).toMatchSnapshot();
        });
      });

      describe('descending', (): void => {
        it('should list first page of competitions with all values', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/competitions').query({
            dir: 'desc',
            sort: 'competition.name'
          });

          expect(status).toEqual(200);
          expect(body).toMatchSnapshot();
        });

        it('should list second page of competitions with all values', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/competitions').query({
            dir: 'desc',
            page: 2,
            sort: 'competition.name'
          });

          expect(status).toEqual(200);
          expect(body).toMatchSnapshot();
        });

        it('should return no content if out of bounds', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/competitions').query({
            dir: 'desc',
            page: 3,
            sort: 'competition.name'
          });

          expect(status).toEqual(204);
          expect(body).toMatchSnapshot();
        });
      });
    });

    describe('with limiting', (): void => {
      it('should list first page of competitions with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/competitions').query({
          limit: 5
        });

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should list second page of competitions with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/competitions').query({
          limit: 5,
          page: 2
        });

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should return no content if out of bounds', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/competitions').query({
          limit: 5,
          page: 4
        });

        expect(status).toEqual(204);
        expect(body).toMatchSnapshot();
      });
    });
  });

  describe('[GET] /competitions/:id', (): void => {
    it('should return a competition', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/competitions/66706c18-4daa-49eb-a857-df12b8e98a8e');

      expect(status).toEqual(200);
      expect(body).toMatchSnapshot();
    });

    it('should return a filtered result', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/competitions/66706c18-4daa-49eb-a857-df12b8e98a8e').query({
        fields: 'competition.name'
      });

      expect(status).toEqual(200);
      expect(body).toMatchSnapshot();
    });

    it('should return a bad request if filtered field does not exist', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/competitions/66706c18-4daa-49eb-a857-df12b8e98a8e').query({
        fields: 'competition.club'
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });

    it('should return not found if competition does not exist', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/competitions/70340225-c90a-4b9b-a979-2d16212acd16');

      expect(status).toEqual(404);
      expect(body).toMatchSnapshot();
    });
  });

  describe('[POST] /competitions', (): void => {
    it('should create a new competition', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.post('/competitions').send({
        name: 'FIFA World Cup'
      });

      id = body.id;

      expect(status).toEqual(201);
      expect(body).toEqual({
        createdAt: expect.any(String),
        id,
        name: 'FIFA World Cup',
        updatedAt: expect.any(String)
      });
    });

    it('should throw an error if a duplicate competition is created', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.post('/competitions').send({
        name: 'FIFA World Cup'
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });

    it('should throw an error if name is not a string', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.post('/competitions').send({
        name: 99
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });
  });

  describe('[PATCH] /competitions/:id', (): void => {
    it('should update a competition', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.patch(`/competitions/${id}`).send({
        name: 'FIFA Club World Cup'
      });

      expect(status).toEqual(204);
      expect(body).toMatchSnapshot();
    });

    it('should throw an error of a competition is updated with a duplicate name', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.patch(`/competitions/${id}`).send({
        name: 'UEFA Nations League'
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });

    it('should return not found if competition does not exist', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.patch('/competitions/88044266-a83a-4e10-b85b-e552790909a1').send({
        name: 'FIFA Club World Cup'
      });

      expect(status).toEqual(404);
      expect(body).toMatchSnapshot();
    });

    it('should throw an error if name is not a string', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.patch(`/competitions/${id}`).send({
        name: 99
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });
  });

  describe('[DELETE] /competitions/:id', (): void => {
    it('should delete a competition', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.delete(`/competitions/${id}`);

      expect(status).toEqual(204);
      expect(body).toMatchSnapshot();
    });

    it('should return not found if competition does not exist', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.delete('/competitions/88044266-a83a-4e10-b85b-e552790909a1');

      expect(status).toEqual(404);
      expect(body).toMatchSnapshot();
    });
  });
});
