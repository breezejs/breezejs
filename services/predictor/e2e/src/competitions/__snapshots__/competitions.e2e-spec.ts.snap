// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Competitions [DELETE] /competitions/:id should delete a competition 1`] = `Object {}`;

exports[`Competitions [DELETE] /competitions/:id should return not found if competition does not exist 1`] = `
Object {
  "error": "Not Found",
  "statusCode": 404,
}
`;

exports[`Competitions [GET] /competitions with filtering should list first page of competitions with all values 1`] = `
Object {
  "items": Array [
    Object {
      "id": "3ae6575c-5857-4f5f-a781-7a0cfba6c2a4",
      "name": "Bundesliga",
    },
    Object {
      "id": "f8a6922f-a7ce-4722-9e67-e5521c219724",
      "name": "FA Cup",
    },
    Object {
      "id": "66706c18-4daa-49eb-a857-df12b8e98a8e",
      "name": "Football League 1",
    },
    Object {
      "id": "ceedef18-0344-40ac-994d-748d963b9fc2",
      "name": "Football League 2",
    },
    Object {
      "id": "c3631875-bad7-4f96-9349-712a89c87f7b",
      "name": "Football League Championship",
    },
    Object {
      "id": "ed530b00-c9fa-49f2-be90-945fc7ca5eca",
      "name": "Football League Cup",
    },
    Object {
      "id": "f0439c4f-206a-48ac-9ff7-db8bad77b8bf",
      "name": "La Liga",
    },
    Object {
      "id": "a6543c15-966a-4d22-85df-edaa11561227",
      "name": "Premier League",
    },
    Object {
      "id": "3c15960b-c2c7-4719-b5bc-19396d33269e",
      "name": "Serie A",
    },
    Object {
      "id": "9463a53e-61cd-4521-8390-3dfd5dcd5fa4",
      "name": "UEFA Champions League",
    },
  ],
  "limit": 10,
  "page": 1,
  "pages": 2,
  "total": 12,
}
`;

exports[`Competitions [GET] /competitions with filtering should list second page of competitions with all values 1`] = `
Object {
  "items": Array [
    Object {
      "id": "5a9e34ef-eae6-417f-b577-0e359eeeeb30",
      "name": "UEFA Europa League",
    },
    Object {
      "id": "ab48a3c7-befd-4411-9ac4-0bd5116ecd99",
      "name": "UEFA Nations League",
    },
  ],
  "limit": 10,
  "page": 2,
  "pages": 2,
  "total": 12,
}
`;

exports[`Competitions [GET] /competitions with filtering should return a bad request if filtered field does not exist 1`] = `
Object {
  "error": "Bad Request",
  "message": "ER_BAD_FIELD_ERROR: Unknown column 'competition.club' in 'field list'",
  "statusCode": 400,
}
`;

exports[`Competitions [GET] /competitions with filtering should return no content if out of bounds 1`] = `Object {}`;

exports[`Competitions [GET] /competitions with limiting should list first page of competitions with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "3ae6575c-5857-4f5f-a781-7a0cfba6c2a4",
      "name": "Bundesliga",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "3c15960b-c2c7-4719-b5bc-19396d33269e",
      "name": "Serie A",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "5a9e34ef-eae6-417f-b577-0e359eeeeb30",
      "name": "UEFA Europa League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "66706c18-4daa-49eb-a857-df12b8e98a8e",
      "name": "Football League 1",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "9463a53e-61cd-4521-8390-3dfd5dcd5fa4",
      "name": "UEFA Champions League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
  ],
  "limit": 5,
  "page": 1,
  "pages": 3,
  "total": 12,
}
`;

exports[`Competitions [GET] /competitions with limiting should list second page of competitions with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "a6543c15-966a-4d22-85df-edaa11561227",
      "name": "Premier League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ab48a3c7-befd-4411-9ac4-0bd5116ecd99",
      "name": "UEFA Nations League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "c3631875-bad7-4f96-9349-712a89c87f7b",
      "name": "Football League Championship",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ceedef18-0344-40ac-994d-748d963b9fc2",
      "name": "Football League 2",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ed530b00-c9fa-49f2-be90-945fc7ca5eca",
      "name": "Football League Cup",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
  ],
  "limit": 5,
  "page": 2,
  "pages": 3,
  "total": 12,
}
`;

exports[`Competitions [GET] /competitions with limiting should return no content if out of bounds 1`] = `Object {}`;

exports[`Competitions [GET] /competitions with no filtering or sorting should list first page of competitions with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "3ae6575c-5857-4f5f-a781-7a0cfba6c2a4",
      "name": "Bundesliga",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "3c15960b-c2c7-4719-b5bc-19396d33269e",
      "name": "Serie A",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "5a9e34ef-eae6-417f-b577-0e359eeeeb30",
      "name": "UEFA Europa League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "66706c18-4daa-49eb-a857-df12b8e98a8e",
      "name": "Football League 1",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "9463a53e-61cd-4521-8390-3dfd5dcd5fa4",
      "name": "UEFA Champions League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "a6543c15-966a-4d22-85df-edaa11561227",
      "name": "Premier League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ab48a3c7-befd-4411-9ac4-0bd5116ecd99",
      "name": "UEFA Nations League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "c3631875-bad7-4f96-9349-712a89c87f7b",
      "name": "Football League Championship",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ceedef18-0344-40ac-994d-748d963b9fc2",
      "name": "Football League 2",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ed530b00-c9fa-49f2-be90-945fc7ca5eca",
      "name": "Football League Cup",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 1,
  "pages": 2,
  "total": 12,
}
`;

exports[`Competitions [GET] /competitions with no filtering or sorting should list second page of competitions with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "f0439c4f-206a-48ac-9ff7-db8bad77b8bf",
      "name": "La Liga",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "f8a6922f-a7ce-4722-9e67-e5521c219724",
      "name": "FA Cup",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 2,
  "pages": 2,
  "total": 12,
}
`;

exports[`Competitions [GET] /competitions with no filtering or sorting should return no content if out of bounds 1`] = `Object {}`;

exports[`Competitions [GET] /competitions with sorting ascending should list first page of competitions with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "3ae6575c-5857-4f5f-a781-7a0cfba6c2a4",
      "name": "Bundesliga",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "f8a6922f-a7ce-4722-9e67-e5521c219724",
      "name": "FA Cup",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "66706c18-4daa-49eb-a857-df12b8e98a8e",
      "name": "Football League 1",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ceedef18-0344-40ac-994d-748d963b9fc2",
      "name": "Football League 2",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "c3631875-bad7-4f96-9349-712a89c87f7b",
      "name": "Football League Championship",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ed530b00-c9fa-49f2-be90-945fc7ca5eca",
      "name": "Football League Cup",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "f0439c4f-206a-48ac-9ff7-db8bad77b8bf",
      "name": "La Liga",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "a6543c15-966a-4d22-85df-edaa11561227",
      "name": "Premier League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "3c15960b-c2c7-4719-b5bc-19396d33269e",
      "name": "Serie A",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "9463a53e-61cd-4521-8390-3dfd5dcd5fa4",
      "name": "UEFA Champions League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 1,
  "pages": 2,
  "total": 12,
}
`;

exports[`Competitions [GET] /competitions with sorting ascending should list second page of competitions with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "5a9e34ef-eae6-417f-b577-0e359eeeeb30",
      "name": "UEFA Europa League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ab48a3c7-befd-4411-9ac4-0bd5116ecd99",
      "name": "UEFA Nations League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 2,
  "pages": 2,
  "total": 12,
}
`;

exports[`Competitions [GET] /competitions with sorting ascending should return no content if out of bounds 1`] = `Object {}`;

exports[`Competitions [GET] /competitions with sorting descending should list first page of competitions with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ab48a3c7-befd-4411-9ac4-0bd5116ecd99",
      "name": "UEFA Nations League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "5a9e34ef-eae6-417f-b577-0e359eeeeb30",
      "name": "UEFA Europa League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "9463a53e-61cd-4521-8390-3dfd5dcd5fa4",
      "name": "UEFA Champions League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "3c15960b-c2c7-4719-b5bc-19396d33269e",
      "name": "Serie A",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "a6543c15-966a-4d22-85df-edaa11561227",
      "name": "Premier League",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "f0439c4f-206a-48ac-9ff7-db8bad77b8bf",
      "name": "La Liga",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ed530b00-c9fa-49f2-be90-945fc7ca5eca",
      "name": "Football League Cup",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "c3631875-bad7-4f96-9349-712a89c87f7b",
      "name": "Football League Championship",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "ceedef18-0344-40ac-994d-748d963b9fc2",
      "name": "Football League 2",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "66706c18-4daa-49eb-a857-df12b8e98a8e",
      "name": "Football League 1",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 1,
  "pages": 2,
  "total": 12,
}
`;

exports[`Competitions [GET] /competitions with sorting descending should list second page of competitions with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "f8a6922f-a7ce-4722-9e67-e5521c219724",
      "name": "FA Cup",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-12-01T00:00:00.000Z",
      "id": "3ae6575c-5857-4f5f-a781-7a0cfba6c2a4",
      "name": "Bundesliga",
      "updatedAt": "2018-12-01T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 2,
  "pages": 2,
  "total": 12,
}
`;

exports[`Competitions [GET] /competitions with sorting descending should return no content if out of bounds 1`] = `Object {}`;

exports[`Competitions [GET] /competitions/:id should return a bad request if filtered field does not exist 1`] = `
Object {
  "error": "Bad Request",
  "message": "ER_BAD_FIELD_ERROR: Unknown column 'competition.club' in 'field list'",
  "statusCode": 400,
}
`;

exports[`Competitions [GET] /competitions/:id should return a competition 1`] = `
Object {
  "createdAt": "2018-12-01T00:00:00.000Z",
  "id": "66706c18-4daa-49eb-a857-df12b8e98a8e",
  "name": "Football League 1",
  "updatedAt": "2018-12-01T00:00:00.000Z",
}
`;

exports[`Competitions [GET] /competitions/:id should return a filtered result 1`] = `
Object {
  "name": "Football League 1",
}
`;

exports[`Competitions [GET] /competitions/:id should return not found if competition does not exist 1`] = `
Object {
  "error": "Not Found",
  "statusCode": 404,
}
`;

exports[`Competitions [PATCH] /competitions/:id should return not found if competition does not exist 1`] = `
Object {
  "error": "Not Found",
  "statusCode": 404,
}
`;

exports[`Competitions [PATCH] /competitions/:id should throw an error if name is not a string 1`] = `
Array [
  Object {
    "children": Array [],
    "constraints": Object {
      "isString": "name must be a string",
    },
    "property": "name",
    "target": Object {
      "name": 99,
    },
    "value": 99,
  },
]
`;

exports[`Competitions [PATCH] /competitions/:id should throw an error of a competition is updated with a duplicate name 1`] = `
Object {
  "error": "Bad Request",
  "message": "ER_DUP_ENTRY: Duplicate entry 'UEFA Nations League' for key 'IDX_4235f333ca8098d3693049ab82'",
  "statusCode": 400,
}
`;

exports[`Competitions [PATCH] /competitions/:id should update a competition 1`] = `Object {}`;

exports[`Competitions [POST] /competitions should throw an error if a duplicate competition is created 1`] = `
Object {
  "error": "Bad Request",
  "message": "ER_DUP_ENTRY: Duplicate entry 'FIFA World Cup' for key 'IDX_4235f333ca8098d3693049ab82'",
  "statusCode": 400,
}
`;

exports[`Competitions [POST] /competitions should throw an error if name is not a string 1`] = `
Array [
  Object {
    "children": Array [],
    "constraints": Object {
      "isString": "name must be a string",
    },
    "property": "name",
    "target": Object {
      "name": 99,
    },
    "value": 99,
  },
]
`;
